### PR TITLE
feat(helm): add --no-hook to helm install and delete

### LINKED ISSUE #932: Add --disable-hooks to helm install/delete
It should be possible to prevent a hook from running.

Specifying `--disable-hooks` should prevent all hooks that would run on that command.

For example, `helm install --disable-hooks foo` would install foo, but would disable both the pre-install and post-install hooks (if they existed).

### PR DESCRIPTION
**Pull request by technosophos**
This includes a substantial bit of unit test improvements. ~~Also, in
order to allow us to tests command line args (which translate to
helm.Option objects), I had to add a new interface to pkg/helm~~ (removed).

Closes #932

### CODE REVIEW

File: pkg/helm/compat.go [hunk 1]
```diff
@@ -16,70 +16,10 @@ limitations under the License.
...
-	rls "k8s.io/helm/pkg/proto/hapi/services"
-)
-
 // These APIs are a temporary abstraction layer that captures the interaction between the current cmd/helm and old
 // pkg/helm implementations. Post refactor the cmd/helm package will use the APIs exposed on helm.Client directly.
 
 // Config is the base configuration
 var Config struct {
 	ServAddr string
 }
```
Comment (fibonacci1729) [inline pkg/helm/compat.go]: Can we drop Config here as well? The ```helm.Host``` option can be used.

File: cmd/helm/install.go [hunk 2]
```diff
@@ -46,70 +47,94 @@ var (
 	installValues string
 	// installRelName is the user-supplied release name.
 	installRelName string
+	// installNoHooks is the flag for controlling hook execution.
+	installNoHooks bool
```
Comment (adamreese) [inline cmd/helm/install.go]: Delete me - globals

**Comment by adamreese** [comment]
after unused globals are deleted, lgtm

**Comment by technosophos** [comment]
Globals deleted, `compat.go` deleted. Waiting for Circle to pass, then will merge.

**Pull request merged**